### ISSUE TITLE
Cannot save any modification for other selection than Default in Configuration Scope

### ISSUE THREAD

**Issue by ADDISON74** [OpenMage/magento-lts#1417]
I just installed OpenMage v20.0.6 from scratch then I accessed the Backend. To reproduce this issue go to **System -> Configuration**.

From left side choose a "Configuration Scope" from the list then press [Save Config] button. Here are the results:

**Default Configuration**

![screenshot1](https://user-images.githubusercontent.com/8360474/105550921-abe68700-5d0a-11eb-8e32-2c5f8b543775.jpg)

**Main Website**

![screenshot2](https://user-images.githubusercontent.com/8360474/105550941-b143d180-5d0a-11eb-9566-705f23ac7936.jpg)

**Default Store View**

![screenshot3](https://user-images.githubusercontent.com/8360474/105550952-b86adf80-5d0a-11eb-9ced-d6a455f7df3b.jpg)

Now let's delete this folder **/app/code/core/Mage** and replace it with the one from OpenMage v20.0.5. Here is the result:

![screenshot4](https://user-images.githubusercontent.com/8360474/105551060-e7815100-5d0a-11eb-88f8-0133ba50b8eb.jpg)

In conclusion "something" in bad happened from 20.0.5 to 20.0.6. There are not many files changed between versions and I don't have time to debug. I will evaluate which files could generate this issue and I hope to find them quick.

BTW reverting this folder to its previous version I was able to use again the Template Hints feature in Frontend.

**Comment by ADDISON74** [comment]
This is the file responsible for this issue:

https://github.com/OpenMage/magento-lts/blob/4e0dc62651e0c983044dde92f30fe81046550c31/app/code/core/Mage/Adminhtml/Block/System/Config/Form.php

lines 628 - 635:

```
        switch ($this->getScope()) {
            case self::SCOPE_DEFAULT:
                return (bool)$field->show_in_default;
            case self::SCOPE_WEBSITES:
                return (bool)$field->show_in_website;
            case self::SCOPE_STORES:
                return (bool)$field->show_in_store;
        }
```

Replace bool with int.

**Comment by colinmollenhour** [comment]
Thanks for the report and finding the location of the bug! This is a pretty serious bug so we should get this merged and released ASAP. Fix in #1418

**Comment by ADDISON74** [comment]
Solved based on the solution @colinmollenhour provided. I will close this issue.
